Specifications/Language/3_Expressions/FunctorApplication.md

```diff
@@ -1,8 +1,37 @@
+Functors are factories that allow to access particular specialization implementations of a callable. Q# currently supports two functors; the `Adjoint` and the `Controlled` function, both of which can be applied to operations that provide the necessary specialization(s). 
+
+The `Controlled` and `Adjoint` functors commute; if `ApplyUnitary` is an operation that supports both functors, then there is no difference between `Controlled Adjoint ApplyUnitary` and `Adjoint Controlled ApplyUnitary`; both have the same type and upon invocation execute the implementation defined for the `controlled adjoint` [specialization](https://github.com/microsoft/qsharp-language/blob/main/Specifications/Language/1_ProgramStructure/4_SpecializationDeclarations.md#specialization-declarations).
+
+## Adjoint functor
+
+If the operation `ApplyUnitary` defines a unitary transformation *U* of the quantum state, `Adjoint ApplyUnitary` accesses the implementation of *U†*. The `Adjoint` functor is its own inverse, since *(U†)† = U* by definition; i.e. `Adjoint Adjoint ApplyUnitary` is the same as `ApplyUnitary`.
+
+The expression `Adjoint ApplyUnitary` is an operation of the same type as `ApplyUnitary`; it has the same argument and return type and supports the same functors. Like any operation, it can be invoked with an argument of suitable type. The following expression will apply the [adjoint specialization](https://github.com/microsoft/qsharp-language/blob/main/Specifications/Language/1_ProgramStructure/4_SpecializationDeclarations.md#specialization-declarations) of `ApplyUnitary` to an argument `arg`:
+```qsharp
+Adjoint ApplyUnitary(arg) 
+```
+
+## Controlled functor
+
+For an operation `ApplyUnitary` that defines a unitary transformation *U* of the quantum state, `Controlled ApplyUnitary` accesses the implementation that applies *U* conditional on all qubits in an array of control qubits being in a state |1⟩. 
+
+The expression `Controlled ApplyUnitary` is an operation with the same return type and [operation characteristics](https://github.com/microsoft/qsharp-language/blob/main/Specifications/Language/4_TypeSystem/OperationsAndFunctions.md#operation-characteristics) as `ApplyUnitary`, meaning it supports the same functors.
+It takes an argument of type `(Qubit[], <TIn>)`, where `<TIn>` should be replace with the argument type of `ApplyUnitary`, taking [singleton tuple equivalence](https://github.com/microsoft/qsharp-language/blob/main/Specifications/Language/4_TypeSystem/SingletonTupleEquivalence.md#singleton-tuple-equivalence) into account. 
+
+| Operation | Argument Type | Controlled Argument Type |
+| --- | --- | --- |
+| X | `Qubit` | `(Qubit[], Qubit)` | 
+| SWAP | `(Qubit, Qubit)` | `(Qubit[], (Qubit, Qubit))` |
```

Comment:
Would it be useful to add some more examples here, perhaps?

Reply:
Do you have a concrete suggestion for an additional example that would be useful?

Reply:
I guess I was thinking of something that has types other than `Qubit`, maybe `AddI : (LittleEndian, LittleEndian) => Unit is Adj + Ctl`?

Reply:
added